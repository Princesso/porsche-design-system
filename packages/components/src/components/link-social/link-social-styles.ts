import { getCss } from '../../utils';
import { addImportantToEachRule } from '../../styles';
import { hostHiddenStyles } from '../../styles/host-hidden-styles';

export const getComponentCss = (): string => {
  return getCss({
    '@global': {
      ':host': addImportantToEachRule({
        ...hostHiddenStyles,
        display: 'inline-block',
        verticalAlign: 'top',
        outline: 0, // custom element is able to delegate the focus
      }),
    },
  });
};
