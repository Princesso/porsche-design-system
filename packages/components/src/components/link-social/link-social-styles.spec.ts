import { getComponentCss } from './link-social-styles';

describe('getComponentCss()', () => {
  it('should return correct css: %s', () => {
    expect(getComponentCss()).toMatchSnapshot();
  });
});
