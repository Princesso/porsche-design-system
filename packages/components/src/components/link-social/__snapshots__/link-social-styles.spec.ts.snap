// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css: %s 1`] = `
":host {
  display: inline-block !important;
  vertical-align: top !important;
  outline: 0 !important;
}
:host([hidden]) {
  display: none !important;
}"
`;
